### ISSUE TITLE
TypeError: choice() takes 1 positional argument but 2 were given

### ISSUE THREAD

**Issue by Phasip** [renatahodovan/grammarinator#16]
After manually fixing issues due to #15 the grammar in https://github.com/chungkwong/fooledit/tree/490c4bc0a4ba6ceec3ac0c4cd1947a54e397ef34/mode.xml/src/main/antlr4/cc/fooledit/editor/text/mode/xml i receive this error:
```
Data: <function XMLUnparser.choice at 0x7f26101f06a8>, ([1, 1],), {}
Test generation failed.
Traceback (most recent call last):
  File "/home/phasip/.local/lib/python3.6/site-packages/grammarinator/generate.py", line 129, in create_new_test
    tree = generator(self.rule, self.max_depth)
  File "/home/phasip/.local/lib/python3.6/site-packages/grammarinator/generate.py", line 168, in generate
    tree = Tree(getattr(self.unparser_cls(unlexer) if rule[0].islower() else unlexer, rule)())
  File "/home/phasip/.local/lib/python3.6/site-packages/grammarinator/runtime/grammarinator.py", line 61, in controlled_fn
    result = fn(obj, *args, **kwargs)
  File "XMLUnparser.py", line 21, in document
    current += self.element()
  File "/home/phasip/.local/lib/python3.6/site-packages/grammarinator/runtime/grammarinator.py", line 61, in controlled_fn
    result = fn(obj, *args, **kwargs)
  File "XMLUnparser.py", line 244, in element
    choice = self.choice([0 if [2, 2][i] > self.unlexer.max_depth else w * self.unlexer.weights.get(('alt_147', i), 1) for i, w in enumerate([1, 1])])
  File "/home/phasip/.local/lib/python3.6/site-packages/grammarinator/runtime/grammarinator.py", line 61, in controlled_fn
    result = fn(obj, *args, **kwargs)
TypeError: choice() takes 1 positional argument but 2 were given
```
Note: The data line comes from a print added by me on grammarinator.py line 60: `print("Data: %s, %s, %s"%(fn,args,kwargs),file=sys.stderr)`

Content of XMLUnparser.py choice function:
```
@depthcontrol
    def choice(self):
        current = self.create_node(UnparserRule(name='choice'))
        current += self.create_node(UnlexerRule(src='('))
        if self.unlexer.max_depth >= 3:
            for _ in self.zero_or_one():
                current += self.s()

        current += self.cp()
        if self.unlexer.max_depth >= 0:
            for _ in self.one_or_more():
                if self.unlexer.max_depth >= 3:
                    for _ in self.zero_or_one():
                        current += self.s()

                current += self.create_node(UnlexerRule(src='|'))
                if self.unlexer.max_depth >= 3:
                    for _ in self.zero_or_one():
                        current += self.s()

                current += self.cp()

        if self.unlexer.max_depth >= 3:
            for _ in self.zero_or_one():
                current += self.s()

        current += self.create_node(UnlexerRule(src=')'))
        return current
    choice.min_depth = 2

```

**Comment by renatahodovan** [comment]
The reason of this failure is that XMLParser contains a rule named `choice` which causes a name collision with the `choice` function provided by the Grammarinator class of the runtime.  I may add some warning to the processor to warn the user in such cases.

**Comment by akosthekiss** [comment]
The introduction of models in 6f43afe3 fixed the problem with `choice`, while 1ba36c8e should fix the problem of name clashes in general.
